OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/RegionAdminControllerTests.cs

```diff
@@ -210,17 +210,36 @@ public async Task Update_WithInvalidModel_ReturnsRequestObjectResult()
         Assert.That(result, Is.InstanceOf<BadRequestObjectResult>());
     }
 
+    [Test]
+    public async Task Update_WorkOnlyWithBaseUserDto()
+    {
+        // Arrange
+        var updateRegionAdminDto = new BaseUserDto();
```

Comment:
Sorry, but I can't agree with this test existence. Probably better delete it.

Details:

The test promises to `_WorkOnlyWithBaseUserDto` but when I change it in test to
`var updateRegionAdminDto = new RegionAdminDto();` 
it doesn't fail. 

If I change it in `UpdateRegionAdminAsync` parameter type, the compiler will error on test code and - see my previous comments - next developer wil just rewrite test to meet the new type.

As for me this checks only one thing - that controller calls service. The same thing that is already verified in 2 tests below.

The initial task - checking absence of "institutionId changing, because it`s not allowed by business rules" is not performed. 

For this check I expect to see in test something like:
```
//Arrange
SeedRepository(new RegionAdminDto() { InstitutionId = oldInstitutionId });
...

//Act
await regionAdminController.Update(new RegionAdminDto() { InstitutionId = otherInstitutionId });

//Assert
var newInstitutionId = GetFromRepository().InstitutionId;
Assert.That(oldInstitutionId, Is.EqualTo(newInstitutionId)); // thus we ensure that it is not changed to otherInstitutionId 
```
Ofc, it's pseudocode as `SeedRepository()` and `GetFromRepository()` are not written and they could be rather big redesign task. It's doable, but I given up to insist on it, and now ask only for this test deletion.

Reply:
@VadymLevkovskyi 
Sorry😓, I want to do this right, not just close PR. Maybe we have misunderstood each other.
Earlier I tried to create such a test as you showed, but in-memory database does not support transactions. We discussed this on meeting, and decided, to try something else. Then i tried to mock send method, but 
> Non-overridable members (here: HttpClient.SendAsync) may not be used in setup / verification expressions.

I can try another option, mock communication service(by changing inheritance with constructor dependency), I don't know any other options

Reply:
Sorry😓, I want to do this right, not just close PR. Maybe we have misunderstood each other.
Earlier I tried to create such a test as you showed, but in-memory database does not support transactions. We discussed this on meeting, and decided, to try something else. Then i tried to mock send method, but

> Non-overridable members (here: HttpClient.SendAsync) may not be used in setup / verification expressions.

**UPD.**
It appeared that i can simply disable that error of in-memory database...😔
Please, check if this test is fits

Reply:
Balissimo! Thank you for being persistent.
There is few more percents of test coverage needed to reach the perfection - please check [Sonar report](https://sonarcloud.io/component_measures?id=ita-social-projects-oos-backend&pullRequest=1272&metric=new_coverage&view=list)

Reply:
TY for your patience. I will work on test coverage percent now